src/components/molecules/Visualizer/utils.ts

```diff
@@ -11,12 +11,16 @@ export function mergeProperty(a: any, b: any) {
   );
 }
 
-export function useOverriddenProperty<T = any>(
-  property: T,
+export function useOverriddenProperty<T extends {}>(
+  property: T | undefined,
 ): [T, (pluginId: string, property: T) => void] {
-  const [{ "": overriddenPropertyCommon, ...overriddenProperty }, setOverrideProperty] = useState<{
+  const [overriddenProperties, setOverrideProperty] = useState<{
     [pluginId: string]: any;
   }>({});
+  const { overriddenPropertyCommon, overriddenProperty } = useMemo(() => {
+    const { "": overriddenPropertyCommon, ...overriddenProperty } = overriddenProperties;
+    return { overriddenPropertyCommon, overriddenProperty };
+  }, [overriddenProperties]);
```

Comment:
This is where it happened: the rest spread syntax always returns a new object. This was causing endless rendering.